evalml/pipelines/components/transformers/encoders/label_encoder.py

```diff
@@ -67,9 +80,13 @@ def transform(self, X, y=None):
         if y is None:
             return X, y
         y_ww = infer_feature_types(y)
-        y_t = self._component_obj.transform(y_ww)
-        y_t = pd.Series(y_t, index=y_ww.index)
-        return X, ww.init_series(y_t)
+        y_unique_values = set(y_ww.unique())
+        if y_unique_values.difference(self.mapping.keys()):
+            raise ValueError(
+                f"y contains previously unseen labels: {y_unique_values.difference(self.mapping.keys())}"
+            )
+        y_t = y_ww.map(self.mapping)
+        return X, ww.init_series(y_t, logical_type="integer")
```

Comment:
Without setting the logical type, we end up with a categorical column which defeats the purpose 😂